### PR TITLE
deny missing docs in ir and opt modules

### PR DESCRIPTION
**Pull request by azarmadr**
i thought this would be easiest way to dive into llhd. and i am only doing it on weekends.

### CODE REVIEW

**Comment by fabianschuiki** [comment]
This is an excellent idea, thanks! Do you want to have this merged now, or will you add additional commits later on?

**Review by fabianschuiki — APPROVED** [review]

**Comment by azarmadr** [comment]
i still have ir module to do
i am new to any open source project involvement
can be merged now i guess?
thanks

**Comment by fabianschuiki** [comment]
In that case, let me change the PR to a draft such that you can continue committing changes. As soon you feel the changes are done and `ir` and `opt` work with `deny(missing_docs)`, we merge the PR :smile:!

**Comment by azarmadr** [comment]
getting a `missing document for a variant` for opcode enum in llhd::ir::inst

File: doc/LANGUAGE.md [hunk 1]
```diff
@@ -459,7 +459,7 @@ The `insert` instruction may be used to change the value of fields of structs, e
 - `index` is the index of the field, element, or bit to be modified. An unsigned integer.
 - `start` is the index of the first element or bit to be modified. An unsigned integer.
 - `length` is the number of elements or bits after `start` to be modified. An unsigned integer.
-- `value` is the value to be assigned o the selected field, elements, or bits. Its type must correspond to a single field, element, or bit in case of the `insert element` variant, or an array or integer of length `length` in case of the `insert slice` variant. A value.
+- `value` is the value to be assigned to the selected field, elements, or bits. Its type must correspond to a single field, element, or bit in case of the `insert element` variant, or an array or integer of length `length` in case of the `insert slice` variant. A value.
```
Comment (fabianschuiki) [inline doc/LANGUAGE.md:462]: Nice catch

File: examples/builder.rs [hunk 2]
```diff
@@ -71,13 +71,13 @@ fn build_entity(name: UnitName) -> UnitData {
...
-    let inp = sig.add_input(llhd::signal_ty(llhd::int_ty(32)));
+    let inp = sig.add_input(llhd::signal_ty(llhd::int_ty(1)));
     let _oup = sig.add_output(llhd::signal_ty(llhd::int_ty(32)));
     let mut ent = UnitData::new(UnitKind::Entity, name, sig);
     {
         let mut builder = UnitBuilder::new_anonymous(&mut ent);
-        let v1 = builder.ins().const_int((32, 42));
-        let v2 = builder.ins().const_int((32, 2));
+        let v1 = builder.ins().const_int((1, 0));
+        let v2 = builder.ins().const_int((1, 1));
```
Comment (fabianschuiki) [inline examples/builder.rs:80]: Cool, thanks for fixing these.

File: src/ir/inst.rs [hunk 3]
```diff
@@ -412,13 +478,15 @@ impl<'a, 'b> InstBuilder<'a, 'b> {
         self.build(data, void_ty())
     }
 
+    /// Creates sig instruction to instantiates a proces or entity
```
Comment (fabianschuiki) [inline src/ir/inst.rs:481]: I think this should read:
```suggestion
    /// Creates sig instruction to allocate a signal
```

**Review by fabianschuiki — CHANGES_REQUESTED** [review]
Just a minor comment with the `sig` instruction doc, then things should be ready to merge. Thanks!

**Comment by fabianschuiki** [comment]
By the way: How about adding `#![deny(missing_docs)]` to `src/ir/mod.rs`? Or at least in those submodules where you have added the missing documentation?

**Comment by azarmadr** [comment]
> 
> 
> By the way: How about adding `#![deny(missing_docs)]` to `src/ir/mod.rs`? Or at least in those submodules where you have added the missing documentation?

regarding that, getting a `missing document for a variant` for opcode enum in `llhd::ir::inst`
should i skip it?

**Comment by fabianschuiki** [comment]
> regarding that, getting a `missing document for a variant` for opcode enum in `llhd::ir::inst`
In that case I would suggest to add `#[allow(missing_docs)]` to the `pub enum Opcode` in order to skip it, as you suggest :+1:!

**Comment by azarmadr** [comment]
i have also added `[allow(missing_docs)]` to some structures which didn't need their fields explaining.
this concludes ir module

**Review by fabianschuiki — APPROVED** [review]
Looks good to me, thanks! Merging :+1:

**Pull request merged**